public/app/plugins/datasource/cloudwatch/datasource.ts

```diff
@@ -490,9 +490,10 @@ export class CloudWatchDatasource extends DataSourceApi<CloudWatchQuery, CloudWa
     };
 
     if (makeReplacements) {
-      requestParams.queries.forEach(
-        query => (query.region = this.replace(this.getActualRegion(this.defaultRegion), scopedVars, true, 'region'))
-      );
+      requestParams.queries.forEach(query => {
+        query.region = this.replace(query.region, scopedVars, true, 'region');
+        query.region = this.getActualRegion(query.region);
```

Comment:
I am not sure what exactly was meant by the former code seems like it was just pretty wrong but maybe I broken some other behavior I don't know about.

@daniellee I realized we may be adding regressions to the cloud watch metrics usecase as we work on single codebase. Is there somebody who could help us with testing that part?

Reply:
@aocenas yes, me, @bergquist and @papagian can help with testing Cloudwatch metrics.